sdk/trace/src/main/java/io/opentelemetry/sdk/trace/internal/ExtendedSpanProcessor.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.sdk.trace.internal;
+
+import io.opentelemetry.api.trace.Span;
+import io.opentelemetry.sdk.trace.ReadWriteSpan;
+import io.opentelemetry.sdk.trace.SpanProcessor;
+
+/**
+ * A specialization of {@link SpanProcessor} providing more extension points.
+ *
+ * <p>Note that this interface is considered experimental and therefore should only be used at the
+ * risk of its methods being changed or removed. If it stabilized, the interface is likely removed
+ * and the methods are promoted to {@link SpanProcessor}.
+ */
+public interface ExtendedSpanProcessor extends SpanProcessor {
+
+  /**
+   * Called just before a {@link io.opentelemetry.api.trace.Span} is ended, if the {@link
```

Comment:
"just before a span is ended" seems wrong. Maybe something like "Called when a Span is ended, but before the SpanProcessor#onEnd is called with an immutable version of the span"